js/sign.js

```diff
@@ -1,43 +1,40 @@
-// // 이메일 입력
-// const querySelector = (selector) => document.querySelector(selector);
-
-// const emailInput = querySelector("#email-input");
-// const emailErrorMessage = querySelector("#email-error-message");
-
-// emailInput.addEventListener("focusout", validateEmailInput);
-
-// function validateEmailInput(emailInput, emailErrorMessage, isValidateEmail, showEmailErrorMessage, hideEmailErrorMessage) {
-//   const email = emailInput.value.trim();
-
-//   if (email === "") {
-//     showEmailErrorMessage(emailErrorMessage, "이메일을 입력해 주세요.");
-//     return;
-//   }
-
-//   if (!isValidateEmail(email)) {
-//     showEmailErrorMessage(emailErrorMessage, "올바른 이메일 주소가 아닙니다.");
-//     return;
-//   }
-
-//   hideEmailErrorMessage(emailInput, emailErrorMessage);
-// }
-
-// function isValidateEmail(email) {
-//   return EMAIL_REGEX.test(email);
-// }
-
-// function showEmailErrorMessage(message) {
-//   emailErrorMessage.textContent = message;
-//   emailErrorMessage.classList.add("show");
-//   emailInput.classList.add("sign-input-error");
-//   emailInput.classList.add("error-focus");
-// }
-
-// function hideEmailErrorMessage() {
-//   emailErrorMessage.textContent = "";
-//   emailErrorMessage.classList.remove("show");
-//   emailInput.classList.remove("sign-input-error");
-//   emailInput.classList.remove("error-focus");
-// }
-
-// export { emailInput, emailErrorMessage, validateEmailInput, isValidateEmail, showEmailErrorMessage, hideEmailErrorMessage };
+// 이메일 유효성 검사
+export function isValidEmail(email) {
+  const EMAIL_REGEX = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6}$/;
+  return EMAIL_REGEX.test(email);
+}
+
+// 비밀번호 유효성 검사
+export function isValidPassword(password) {
+  const PASSWORD_REGEX = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;
+  return PASSWORD_REGEX.test(password);
+}
+
+// 오류 메시지 표시
+export function showErrorMessage(inputElement, messageElement, message) {
+  messageElement.textContent = message;
+  messageElement.classList.add("show");
+  inputElement.classList.add("sign-input-error");
+  inputElement.classList.add("error-focus");
+}
+
+// 오류 메시지 숨기기
+export function hideErrorMessage(inputElement, messageElement) {
+  messageElement.textContent = "";
+  messageElement.classList.remove("show");
+  inputElement.classList.remove("sign-input-error");
+  inputElement.classList.remove("error-focus");
+}
+
+// 눈 아이콘 클릭 핸들러
+export function togglePasswordVisibility(inputElement, eyeOffButton, eyeOnButton) {
+  if (inputElement.type === "password") {
+    inputElement.type = "text";
+    eyeOffButton.classList.add("hide");
+    eyeOnButton.classList.remove("hide");
+  } else {
+    inputElement.type = "password";
+    eyeOffButton.classList.remove("hide");
+    eyeOnButton.classList.add("hide");
+  }
```

Comment:
이 부분은 사소하긴 한데요!
지금처럼 코드가 작성되어 있으면 inputElement의 type이 text가 아니더라도, else문이 실행될 수 있는데요! (inputElemet의 type이 password만 아니면 무조건 else문이 실행되기 때문)

만약 개발자가 코드를 잘못 작성하면 버그가 발생할 수도 있어서 이런식으로 명시적으로 if문으로 처리를 해주면 어떨까 싶었어요!
```suggestion
  if (inputElement.type === "password") {
    inputElement.type = "text";
    eyeOffButton.classList.add("hide");
    eyeOnButton.classList.remove("hide");
    return;
  } 
  
  if (inputElement.type === 'text') {
    inputElement.type = "password";
    eyeOffButton.classList.remove("hide");
    eyeOnButton.classList.add("hide");
  }
```